ExampleMod/Projectiles/ExampleLastPrismHoldout.cs

```diff
@@ -0,0 +1,216 @@
+		// When first used, the Prism consumes mana once every MaxManaConsumptionDelay frames.
+		// Every time mana is consumed, the pace becomes one frame faster, meaning mana consumption smoothly increases.
+		// When capped out, the Prism consumes mana once every MinManaConsumptionDelay frames.
+		private const float MaxManaConsumptionDelay = 15f;
+		private const float MinManaConsumptionDelay = 5f;
+
+		// This property encloses the internal AI variable projectile.ai[0]. It makes the code easier to read.
+		private float FrameCounter {
+			get => projectile.ai[0];
+			set => projectile.ai[0] = value;
+		}
+
+		// This property encloses the internal AI variable projectile.ai[1].
+		private float NextManaFrame {
+			get => projectile.ai[1];
+			set => projectile.ai[1] = value;
+		}
+
+		// This property encloses the internal AI variable projectile.localAI[0].
+		// localAI is not automatically synced over the network, but that does not cause any problems in this case.
+		private float ManaConsumptionRate {
+			get => projectile.localAI[0];
+			set => projectile.localAI[0] = value;
+		}
+
+		public override void SetStaticDefaults()
+		{
+			DisplayName.SetDefault("Example Last Prism");
+			Main.projFrames[projectile.type] = 5;
+		}
+
+		public override void SetDefaults()
+		{
+			projectile.CloneDefaults(ProjectileID.LastPrism);
+		}
+
+		public override void AI()
+		{
+			Player player = Main.player[projectile.owner];
+			Vector2 rrp = player.RotatedRelativePoint(player.MountedCenter, true);
+
+			// Update the Prism's damage every frame so that it is dynamically affected by Mana Sickness.
+			UpdateDamageForManaSickness(player);
+
+			// Update the frame counter.
+			FrameCounter += 1f;
+
+			// Update the Prism's animation, with the animation accelerating as the Prism charges.
+			projectile.frameCounter++;
+			int framesPerAnimationUpdate = FrameCounter >= MaxCharge ? 2 : FrameCounter >= (MaxCharge * 0.66f) ? 3 : 4;
+			if (projectile.frameCounter >= framesPerAnimationUpdate)
+			{
+				projectile.frameCounter = 0;
+				if (++projectile.frame >= 5)
+					projectile.frame = 0;
+			}
+
+			// Make sound intermittently while the Prism is in use.
+			if (projectile.soundDelay <= 0)
+			{
+				projectile.soundDelay = SoundInterval;
+				// On the very first frame, the sound playing is skipped. This way it doesn't overlap the starting hiss sound.
+				if (FrameCounter > 1f)
+					Main.PlaySound(SoundID.Item15, projectile.position);
+			}
+
+			// Update the Prism's position in the world and relevant variables of the player holding it.
+			UpdatePlayerVisuals(player, rrp);
+
+			// Update the Prism's behavior: project beams on frame 1, consume mana, and despawn if out of mana.
+			if (projectile.owner == Main.myPlayer)
+			{
+				// Slightly re-aim the Prism every frame so that it gradually sweeps to point towards the mouse.
+				UpdateAim(rrp, player.inventory[player.selectedItem].shootSpeed);
```

Comment:
`player.inventory[player.selectedItem]` is `player.HeldItem`, same a few lines below